### ISSUE TITLE
Embedded system 

### ISSUE THREAD

**Issue by raaka1** [tbennun/cudnn-training#15]
Hey i am testing it on an Embedded system.

cudnn-training$ ./trainlenet 
```
Reading input data
Done. Training dataset size: 60000, Test dataset size: 10000
Batch size: 64, iterations: 1000
Preparing dataset
Training...
Iteration time: 12.190692 ms
Classification result: 8.43% error (used 10000 images)
```

am i supposed to give image as an input ? How is the result displayed.

**Comment by tbennun** [comment]
This sample only trains the network on MNIST. If you want inference, you can easily implement it within lenet.cu by loading the image to memory (normalized to [0,1]), creating a TrainingContext with your preferred batch size (1 for a single image), and calling ForwardPropagation. See [Line 989](https://github.com/tbennun/cudnn-training/blob/master/lenet.cu#L989) for more information.

**Comment by raaka1** [comment]
Is the Classification result: 8.43% error (used 10000 images) normal ?

**Comment by tbennun** [comment]
By default it runs a single epoch (a bit more than that) and that's not enough to get good classification results, so you may want to run more iterations. Also, this is a relatively simple architecture, others can learn to be as low as 0.2% classification error.

Keep in mind that this project is meant to be a simple code example for CUDNN training, not a state-of-the-art deep learning framework.

**Comment by raaka1** [comment]
Its an Nvidia Embedded system, not keen on running a full framework.Your code is fine.

**Comment by raaka1** [comment]
I just want to detect numbers from camera input.

after raising learning rate
`DEFINE_double(learning_rate, 0.06, "Base learning rate");`

```
Reading input data
Done. Training dataset size: 60000, Test dataset size: 10000
Batch size: 64, iterations: 1000
Preparing dataset
Training...
Iteration time: 12.336006 ms
Classification result: 2.58% error (used 10000 images)
```

**Comment by tbennun** [comment]
Good. It is customary to decrease the learning rate as the iterations increase, so maybe after 1000 iterations decrease learning rate to 0.006.

Anyway, training, especially on embedded systems, is meant to be performed offline (saving the network weights after a satisfactory error rate). The inference stage (i.e., looking at an actual number) is then performed online with the preloaded weights.

**Comment by raaka1** [comment]
Is it possible to provide, code sample adding image to program ? for detection

**Comment by tbennun** [comment]
I don't think it is relevant. This sample is meant for demonstration purposes on how to use CUDNN for training. Loading images is completely unrelated to that.

It's not that hard, you find a library that will load your image to memory (e.g. OpenCV), then you load and resize it to 28x28 pixels with floating point values between 0 and 1, create a TrainingContext with a batch size of 1 (after training of course), and pass the buffer through the ForwardPropagation method of TrainingContext.

I'm closing this issue since it's unrelated to the sample itself. Good luck! :)
